cmd/wallet/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+
+	"github.com/hashcloak/Meson-wallet-demo/pkg/ethers"
+	"github.com/hashcloak/Meson-client/pkg/client"
+	"github.com/katzenpost/client/config"
+)
+
+func main() {
+	cfgFile := flag.String("c", "katzenpost.toml", "Path to the server config file")
+	ticker := flag.String("t", "", "Ticker")
+	chainID := flag.Int("chain", 1, "Chain ID for specific ETH-based chain")
+	service := flag.String("s", "", "Service Name")
+	rawTransactionBlob := flag.String("rt", "", "Raw Transaction blob to send over the network")
+	privKey := flag.String("pk", "", "Private key used to sign the txn")
+	rpcEndpoint := flag.String("rpc", "http://172.28.1.10:9545", "Ethereum rpc endpoint")
+	flag.Parse()
+
+	cfg, err := config.LoadFile(*cfgFile)
+	if err != nil {
+		panic(err)
+	}
+
+    if *rawTransactionBlob == "" {
+		if *privKey == "" {
+			panic("must specify a transaction blob in hex or a private key to sign a txn")
+		}
+		rawTransactionBlob, err = produceSignedRawTxn(privKey, rpcEndpoint, chainID)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+    c, err := client.New(cfg, *service)
+    if err != nil {
+        panic(err)
+    }
+
+    c.Start()
+    reply, err := c.SendRawTransaction(rawTransactionBlob, chainID, ticker)
+    if err != nil {
+        panic(err)
+    }
+
+    fmt.Sprintf("Reply from the provider: %s", reply)
+    c.Stop()
+	
+
+}
+
+func produceSignedRawTxn(pk *string, rpcEndpoint *string, chainID *int) (*string, error) {
```

Comment:
I know this is a helper but shouldn't this be in the `ethers` package?

Reply:
Yeah, I thought about that... I wanted to keep the contents of `func produceSignedRawTxn` seperate from the rest of the main code just in case the wallet usage changes. Specifically if the way nonces and gas prices are obtained. Otherwise this is most likely a case of premature optimization...

I will work on this tomorrow morning.